projects/plugins/jetpack/extensions/blocks/premium-content/premium-content.php

```diff
@@ -25,21 +25,26 @@
  * registration if we need to.
  */
 function register_block() {
-	// Determine required `context` key based on Gutenberg version.
-	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
-	$provides   = $deprecated ? 'providesContext' : 'provides_context';
 
-	Blocks::jetpack_register_block(
-		__DIR__,
-		array(
-			'render_callback' => __NAMESPACE__ . '\render_block',
-			$provides         => array(
-				'premium-content/planId'  => 'selectedPlanId', // Deprecated.
-				'premium-content/planIds' => 'selectedPlanIds',
-				'isPremiumContentChild'   => 'isPremiumContentChild',
-			),
-		)
-	);
+	require_once JETPACK__PLUGIN_DIR . '/modules/memberships/class-jetpack-memberships.php';
+	if ( \Jetpack_Memberships::is_enabled_jetpack_recurring_payments() ) {
+
+		// Determine required `context` key based on Gutenberg version.
+		$deprecated = function_exists( 'gutenberg_get_post_from_context' );
```

Comment:
This makes me realize this may be a good time to get rid of this, since `gutenberg_get_post_from_context` no longer exists. This should probably happen in another PR though, since it's unrelated to the issue at hand.

Reply:
I wasn't sure what it was exactly. Happy to remove it here as well, as it is a very small update...

Reply:
PR ready in https://github.com/Automattic/jetpack/pull/39319